rotkehlchen/api/rest.py

```diff
@@ -3196,3 +3196,93 @@ def purge_gitcoin_grant_data(self, grant_id: Optional[int]) -> Response:
             self.rotkehlchen.data.db, self.rotkehlchen.msg_aggregator,
         ).delete_gitcoin_ledger_actions(grant_id)
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
+
+    def _edit_manual_price(self, historical_price: HistoricalPrice) -> Dict[str, Any]:
+        # pylint: disable=no-self-use
+        edited = GlobalDBHandler().edit_manual_price(historical_price)
+        if edited:
+            return OK_RESULT
+        return {
+            'result': None,
+            'message': 'Failed to edit historical price entry.',
+            'status_code': HTTPStatus.BAD_REQUEST,
+        }
+
+    def edit_manual_price(
+        self,
+        from_asset: Asset,
+        to_asset: Asset,
+        price: Price,
+        timestamp: Timestamp,
+        async_query: bool,
```

Comment:
From the entire logic of this PR get rid of `async_query`. No remote calls are happening. It's all just querying the local DB. Keep it simple.